jscomp/others/js_promise2.res

```diff
@@ -16,3 +17,45 @@ let catch: (promise<'a>, error => promise<'a>) => promise<'a> = %raw(`
 
 @deprecated("Use then instead")
 let then_ = then
```

Comment:
I think this can be removed again now.

Reply:
Yes I guess I'm removing it.
In case of update from `Js.Promise`, it's a type error anyway. Without this, it's a "do you mean then`? Followed by a type error. Better do all the fixes at once.